agent/engine/dockerclient/versions.go

```diff
@@ -31,98 +23,28 @@ const (
 	Version_1_21 DockerVersion = "1.21"
 	Version_1_22 DockerVersion = "1.22"
 	Version_1_23 DockerVersion = "1.23"
-
-	defaultVersion = Version_1_17
+	Version_1_24 DockerVersion = "1.24"
+	Version_1_25 DockerVersion = "1.25"
+	Version_1_26 DockerVersion = "1.26"
+	Version_1_27 DockerVersion = "1.27"
+	Version_1_28 DockerVersion = "1.28"
 )
 
-var supportedVersions []DockerVersion
-
-func init() {
-	supportedVersions = []DockerVersion{
+// getKnownAPIVersions returns all of the API versions that we know about.
+// It doesn't care if the version is supported by docker or ECS agent
+func getKnownAPIVersions() []DockerVersion {
+	return []DockerVersion{
 		Version_1_17,
 		Version_1_18,
 		Version_1_19,
 		Version_1_20,
 		Version_1_21,
 		Version_1_22,
 		Version_1_23,
+		Version_1_24,
+		Version_1_25,
+		Version_1_26,
+		Version_1_27,
+		Version_1_28,
```

Comment:
Yeah! Its for future use cases. In the interface for the factory, we only currently expose what is supported by both agent and docker.

@samuelkarp had mentioned to me that it would be useful to understand all of the API versions that the host's docker daemon supports even if we don't leverage it to the main engine code.